cyipopt/cython/ipopt_wrapper.pyx

```diff
@@ -858,16 +858,20 @@ cdef Bool objective_cb(Index n,
                        Bool new_x,
                        Number* obj_value,
                        UserDataPtr user_data
-                       ):
-
-    log(b"objective_cb", logging.INFO)
-
-    cdef object self = <object>user_data
+                       ) noexcept:
+    cdef Problem self
     cdef Index i
-    cdef np.ndarray[DTYPEd_t, ndim=1] _x = np.zeros((n,), dtype=DTYPEd)
-    for i in range(n):
-        _x[i] = x[i]
+    cdef np.ndarray[DTYPEd_t, ndim=1] _x
+
     try:
+        log(b"objective_cb", logging.INFO)
+
+        self = <Problem>user_data
+        _x = np.zeros((n,), dtype=DTYPEd)
+
+        for i in range(n):
+            _x[i] = x[i]
```

Comment:
How will these new lines raise a `CyIpoptEvalutionError`? It isn't clear why you've moved them in this try/except.